### PR TITLE
Add entitlement.properties templating. 

### PR DESCRIPTION
**Pull request by hasinidilanka**
**Proposed Changes**

* Add `entitlement.properties` templating.
* Add related configurations to `default.json` file.

### CODE REVIEW

File: modules/distribution/src/repository/resources/conf/templates/repository/conf/identity/entitlement.properties [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+JSON.Shorten.Form.Enabled={{identity.entitlement.JSON_shorten_form_enabled}}
+
+
+#PDP.Extensions.Extension.1=your.extension.class.name
+
+
+#PDP.Policy.Store=org.wso2.carbon.identity.entitlement.policy.store.CarbonRegistryPolicyStore
+
+PIP.AttributeDesignators.Designator.1={{identity.entitlement.policy_point.pip.attribute_designators.designator_1}}
+PIP.AttributeDesignators.Designator.2={{identity.entitlement.policy_point.pip.attribute_designators.designator_2}}
```
Comment (pulasthi7) [inline modules/distribution/src/repository/resources/conf/templates/repository/conf/identity/entitlement.properties]: Better to loop this

File: modules/distribution/src/repository/resources/conf/templates/repository/conf/identity/entitlement.properties [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+PIP.ResourceFinders.Finder.1={{identity.entitlement.policy_point.pip.resource_finder}}
+
+PAP.Entitlement.Data.Finder.1={{identity.entitlement.policy_point.pap.entitlement_data_Finder}}
+PAP.Policy.Publisher.Module.1={{identity.entitlement.policy_point.pap.policy_publisher_module}}
+#PAP.Policy.Post.Publisher.Module.1=
+#PAP.Policy.Publisher.Verification.Handler=
+PAP.Policy.Version.Module={{identity.entitlement.policy_point.pap.policy_version_module}}
+PAP.Status.Data.Handler.1={{identity.entitlement.policy_point.pap.status_data_handler}}
+
+PDP.Policy.Finder.1={{identity.entitlement.policy_point.pdp.policy_finder}}
```
Comment (pulasthi7) [inline modules/distribution/src/repository/resources/conf/templates/repository/conf/identity/entitlement.properties]: Use loops for the places which can have multiple items.

File: modules/distribution/src/repository/resources/conf/templates/repository/conf/identity/entitlement.properties (hunk 1 above)
Comment (hasinidilanka) [reply modules/distribution/src/repository/resources/conf/templates/repository/conf/identity/entitlement.properties]: Fixed.

File: modules/distribution/src/repository/resources/conf/templates/repository/conf/identity/entitlement.properties (hunk 2 above)
Comment (hasinidilanka) [reply modules/distribution/src/repository/resources/conf/templates/repository/conf/identity/entitlement.properties]: Fixed.

File: modules/distribution/src/repository/resources/conf/default.json [hunk 3]
```diff
@@ -190,5 +190,67 @@
...
+
+
+  "identity.entitlement.policy_point": {
+    "pdp": {
+      "enabled": false,
+      "default_caching_interval": "300",
+      "caching": {
+        "decision_caching": {
+          "enabled": true,
+          "caching_interval": "{{identity.entitlement.policy_point.pdp.default_caching_interval}}"
```
Comment (pulasthi7) [inline modules/distribution/src/repository/resources/conf/default.json]: This should be "$conf{identity.entitlement.policy_point.pdp.default_caching_interval}"

Comment (hasinidilanka) [reply modules/distribution/src/repository/resources/conf/default.json]: Fixed by [70dd62b](https://github.com/wso2/product-is/pull/4513/commits/70dd62bff4e55f41924bfdb5daf427f11694d2f6)

File: modules/distribution/src/repository/resources/conf/default.json [hunk 4]
```diff
@@ -190,5 +190,50 @@
 
 
   "identity.auth_framework.endpoint.tenant_list_enabled" : false,
-  "identity.auth_framework.endpoint.mutual_ssl_manager_enabled" : true
+  "identity.auth_framework.endpoint.mutual_ssl_manager_enabled" : true,
+
```
Comment (GDRDABARERA) [inline modules/distribution/src/repository/resources/conf/default.json]: unnecessary new lines

Comment (hasinidilanka) [reply modules/distribution/src/repository/resources/conf/default.json]: These files were moved to the component level and refactored.

**Pull request merged**